File: python-version/LC2843/count_symmetric_integers.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        ans = 0
        for i in range(low, high+1):
            num = str(i)
            n = len(num)
            if n & 1 == 1:
                continue
            judge = 0
            for j in range(n//2):
                judge += ord(num[j])
            for j in range(n//2, n):
                judge -= ord(num[j])
            ans += 1 if judge == 0 else 0
        return ans

File: go-version/leetcode/LC2843/count_symmetric_integers.go
package LC2843

import "strconv"

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/4/11 下午9:54
 * @FilePath: count_symmetric_integers
 * @Description: 遍历数字进行判断
 */

func countSymmetricIntegers(low, high int) int {
	var ans int
	for i := low; i <= high; i++ {
		str := strconv.Itoa(i)
		n := len(str)
		if n&1 == 1 {
			continue
		}
		judge := 0
		for j := 0; j < n/2; j++ {
			judge += int(str[j])
		}
		for j := n / 2; j < n; j++ {
			judge -= int(str[j])
		}
		if judge == 0 {
			ans++
		}
	}
	return ans
}
